### ISSUE TITLE
BayesSearchCV optimized model predicted test set better than train set

### ISSUE THREAD

**Issue by Marc0935** [scikit-optimize/scikit-optimize#780]
Hi skopt developers,

I performed hyperparameter tuning for ligthGBM classification model using BayesSearchCV() function. 

The complete dataset was spitted into train set and test set. I fit the opt=BayesSearchCV(CV=10).fit(TrainX,TrainY) function with train set and then evaluate the mode performance on test set using opt.score(TestX,TestY). The scoring function used for optimization is balanced accuracy.

I found that the test set scores are slightly 2-3% better than the train set ones for several imbalanced datasets, when comparing the train test results.

I wish know whether it is normal to see the optimized model predicts test set slight better than train set? Do you have any explanation for my findings?

Thanks!

**Comment by ogencoglu** [comment]
You can encounter such a phenomena due to several reasons including aleatoric uncertainty in your data, data size, the way you split etc., especially for imbalanced stuff. This is an issue of whether your test set is sampled from the same distribution of the training set and with how much certainty one can claim that, not much about function minimization.
